### PR TITLE
Contraint batchUpdatePartitionAsync request size when talking to AWS Glue

### PR DESCRIPTION
**Pull request by losipiuk**
Note: PR from trinodb/trino so AWS Glue tests are run

### CODE REVIEW

**Review by hashhar — APPROVED** [review]
Looks good to me.

File: plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/glue/GlueHiveMetastore.java [hunk 1]
```diff
@@ -419,17 +420,22 @@ private void updatePartitionStatisticsBatch(Table table, Map<String, Function<Pa
...
+                    .withTableName(table.getTableName())
+                    .withEntries(partitionUpdateRequestsPartition)));
+        });
+
         try {
             // Update column statistics
             columnStatisticsProvider.updatePartitionStatistics(columnStatisticsUpdates.build());
             // Don't block on the batch update call until the column statistics have finished updating
-            getFutureValue(updatePartitionResult);
+            partitionUpdateRequestsFutures.forEach(MoreFutures::getFutureValue);
```
Comment (findepi) [inline plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/glue/GlueHiveMetastore.java:438]: preexisting, but is it OK to send `updatePartitionStatistics` even before `batchUpdatePartitionAsync` is finished?
ie is there no validation on the Glue side?

**Review by findepi — APPROVED** [review]

File: plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/glue/GlueHiveMetastore.java (hunk 1 above)
Comment (losipiuk) [reply plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/glue/GlueHiveMetastore.java:438]: Good point. IDK I assume there can be a validation that e.g. NDV is >= number of rows.
On the other hand there could also be validation that when you call `batchUpdatePartitionAsync` that it is in line with column statistics already present in Glue.

Comment (losipiuk) [reply plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/glue/GlueHiveMetastore.java:438]: I clicked through fields in https://docs.aws.amazon.com/glue/latest/webapi/API_UpdateColumnStatisticsForPartition.html and they do not document any smart validation rules. Just that values need to be >=0

**Pull request merged**